paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziCallback.kt

```diff
@@ -88,11 +89,26 @@ internal class PaparazziCallback(
     name: String,
     constructorSignature: Array<Class<*>>,
     constructorArgs: Array<Any>
+  ): Any? = createNewInstance(name, constructorSignature, constructorArgs)
+
+  override fun loadClass(
+    name: String,
+    constructorSignature: Array<Class<*>>,
+    constructorArgs: Array<Any>
   ): Any? {
-    val viewClass = Class.forName(name)
-    val viewConstructor = viewClass.getConstructor(*constructorSignature)
-    viewConstructor.isAccessible = true
-    return viewConstructor.newInstance(*constructorArgs)
+    // RecyclerView.Adapter is an abstract class, but its instance is needed for RecyclerView to work correctly.
+    // So, when LayoutLib asks for its instance, we define a new class which extends the Adapter class.
+    // We check whether the class being loaded is the support or the androidx one and use the appropriate adapter that references to the
+    // right namespace.
```

Comment:
@JakeWharton intercepting the call was too much of a hassle.  so this copies the AS implementation with a dummy adapter (during inflation code paths only, in BridgeInflater), until the actual adapter eventually gets wired anyway.